File: tecnicell.client/src/app/Interfaces/business/Battery.ts
import { Branch } from "./Branch";
import { Sale } from "./Sale";

export interface Battery {
    batteryCode:      string;
    name:             string;
    brand:            string;
    quantity:         number;
    salePrice:        number;
    warranty:         number;
    batteryHistories: BatteryHistory[];
    brandNavigation:  BatteryBrand;
}

export interface BatteryHistory {
    batteryCode:        string;
    date:               Date;
    actionHistory:      string;
    toBranch:           string;
    description:        string;
    quantity:           number;
    saleCode:           string;
    saleCodeNavigation: Sale;
    toBranchNavigation: Branch;
}

export interface BatteryBrand {
    name:        string;
    description: string;
}

File: Tecnicell.Server/Models/Entity/Battery.cs
using System;
using System.Collections.Generic;

namespace Tecnicell.Server.Models.Entity;

public partial class Battery
{
    public string BatteryCode { get; set; } = null!;

    public string? Name { get; set; }

    public string? Brand { get; set; }

    public decimal? SalePrice { get; set; }

    public string? ImageCode { get; set; }

    public int? Warranty { get; set; }

    public virtual ICollection<BatteryHistory> BatteryHistories { get; set; } = new List<BatteryHistory>();

    public virtual Brand? BrandNavigation { get; set; }

    public virtual Image? ImageCodeNavigation { get; set; }
}
